diff --git a/microsoft/resnet-50/model/graph_0/main.py b/microsoft/resnet-50/model/graph_0/main.py
index 9b1dbb9..85b65c5 100644
--- a/microsoft/resnet-50/model/graph_0/main.py
+++ b/microsoft/resnet-50/model/graph_0/main.py
@@ -1,3 +1,5 @@
+import time
+
 import ttnn
 import torch
 import utils
@@ -44,9 +46,18 @@ def test_main():
 
     device = open_device()
     try:
-        load_activations_for__main_0 = load_activations_for__main(device)
         model = ModelTTNN(device)
-        _main_0 = model(load_activations_for__main_0)
+
+        for i in range(3):
+            load_activations_for__main_0 = load_activations_for__main(device)
+            batch_size = load_activations_for__main_0[0].shape[0]
+            start_time = time.perf_counter()
+            _main_0 = model(load_activations_for__main_0)
+            ttnn.synchronize_device(device)
+            end_time = time.perf_counter()
+            time_taken = end_time - start_time
+            fps = batch_size / time_taken
+            print(f"Iteration {i + 1}: Time taken: {time_taken:.4f}s, FPS: {fps:.2f}")
 
         ttnn_output = ttnn.to_torch(_main_0[0]).reshape(1, 1000).to(torch.float32)
         golden_output = model_pt.main().logits.reshape(1, 1000).to(torch.float32)
