diff --git a/microsoft/resnet-50/model/graph_0/model_ttnn.py b/microsoft/resnet-50/model/graph_0/model_ttnn.py
index f8d731a..2c155b9 100644
--- a/microsoft/resnet-50/model/graph_0/model_ttnn.py
+++ b/microsoft/resnet-50/model/graph_0/model_ttnn.py
@@ -14,13 +14,58 @@ class ModelTTNN(LightweightModule):
         self.weights = params.load_weights_for__main_from_state_dict()
         self.weights = consteval.run_consteval(self.weights, device)
 
+        self.embedder = ResNetEmbeddings(device, self.weights)
+        self.stage0_layer0 = ResNetBottleNeckLayer_S0_L0(device, self.weights)
+        self.stage0_layer1 = ResNetBottleNeckLayer_S0_L1(device, self.weights)
+        self.stage0_layer2 = ResNetBottleNeckLayer_S0_L2(device, self.weights)
+        self.stage1_layer0 = ResNetBottleNeckLayer_S1_L0(device, self.weights)
+        self.stage1_layer1 = ResNetBottleNeckLayer_S1_L1(device, self.weights)
+        self.stage1_layer2 = ResNetBottleNeckLayer_S1_L2(device, self.weights)
+        self.stage1_layer3 = ResNetBottleNeckLayer_S1_L3(device, self.weights)
+        self.stage2_layer0 = ResNetBottleNeckLayer_S2_L0(device, self.weights)
+        self.stage2_layer1 = ResNetBottleNeckLayer_S2_L1(device, self.weights)
+        self.stage2_layer2 = ResNetBottleNeckLayer_S2_L2(device, self.weights)
+        self.stage2_layer3 = ResNetBottleNeckLayer_S2_L3(device, self.weights)
+        self.stage2_layer4 = ResNetBottleNeckLayer_S2_L4(device, self.weights)
+        self.stage2_layer5 = ResNetBottleNeckLayer_S2_L5(device, self.weights)
+        self.stage3_layer0 = ResNetBottleNeckLayer_S3_L0(device, self.weights)
+        self.stage3_layer1 = ResNetBottleNeckLayer_S3_L1(device, self.weights)
+        self.stage3_layer2 = ResNetBottleNeckLayer_S3_L2(device, self.weights)
+        self.classifier = ResNetClassifier(device, self.weights)
+
     def forward(self, activations):
-        args_0 = activations[0]
-        utils_DeviceGetter_get_device_0 = self.device
+        x = activations[0]
+        x = self.embedder(x)
+        x = self.stage0_layer0(x)
+        x = self.stage0_layer1(x)
+        x = self.stage0_layer2(x)
+        x = self.stage1_layer0(x)
+        x = self.stage1_layer1(x)
+        x = self.stage1_layer2(x)
+        x = self.stage1_layer3(x)
+        x = self.stage2_layer0(x)
+        x = self.stage2_layer1(x)
+        x = self.stage2_layer2(x)
+        x = self.stage2_layer3(x)
+        x = self.stage2_layer4(x)
+        x = self.stage2_layer5(x)
+        x = self.stage3_layer0(x)
+        x = self.stage3_layer1(x)
+        x = self.stage3_layer2(x)
+        return self.classifier(x)
+
+
+
+class ResNetEmbeddings(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_to_layout_0 = ttnn.to_layout(
-            args_0, ttnn.Layout.TILE, None, memory_config=None
+            x, ttnn.Layout.TILE, None, memory_config=None
         )
-        ttnn.deallocate(args_0, False)
+        ttnn.deallocate(x, False)
         ttnn_permute_0 = ttnn.permute(
             ttnn_to_layout_0,
             [0, 2, 3, 1],
@@ -41,7 +86,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_0 = ttnn.conv2d(
             input_tensor=ttnn_reshape_0,
             weight_tensor=self.weights["resnet.embedder.embedder.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=3,
             out_channels=64,
             batch_size=8,
@@ -110,10 +155,20 @@ class ModelTTNN(LightweightModule):
             config_tensor_in_dram=True,
         )
         ttnn.deallocate(ttnn_conv2d_0, False)
+        return ttnn_max_pool2d_0
+
+
+
+class ResNetBottleNeckLayer_S0_L0(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_conv2d_1 = ttnn.conv2d(
-            input_tensor=ttnn_max_pool2d_0,
+            input_tensor=x,
             weight_tensor=self.weights["resnet.encoder.stages.0.layers.0.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=64,
             out_channels=64,
             batch_size=8,
@@ -155,7 +210,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_2 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_1,
             weight_tensor=self.weights["resnet.encoder.stages.0.layers.0.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=64,
             out_channels=64,
             batch_size=8,
@@ -197,7 +252,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_3 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_2,
             weight_tensor=self.weights["resnet.encoder.stages.0.layers.0.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=64,
             out_channels=256,
             batch_size=8,
@@ -236,7 +291,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn_to_memory_config_0 = ttnn.to_memory_config(
-            ttnn_max_pool2d_0,
+            x,
             ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
                 ttnn.BufferType.L1,
@@ -252,11 +307,11 @@ class ModelTTNN(LightweightModule):
                 ),
             ),
         )
-        ttnn.deallocate(ttnn_max_pool2d_0, False)
+        ttnn.deallocate(x, False)
         ttnn_conv2d_4 = ttnn.conv2d(
             input_tensor=ttnn_to_memory_config_0,
             weight_tensor=self.weights["resnet.encoder.stages.0.layers.0.shortcut.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=64,
             out_channels=256,
             batch_size=8,
@@ -333,10 +388,20 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_0, False)
+        return ttnn_relu_0
+
+
+
+class ResNetBottleNeckLayer_S0_L1(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_conv2d_5 = ttnn.conv2d(
-            input_tensor=ttnn_relu_0,
+            input_tensor=x,
             weight_tensor=self.weights["resnet.encoder.stages.0.layers.1.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=64,
             batch_size=8,
@@ -378,7 +443,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_6 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_5,
             weight_tensor=self.weights["resnet.encoder.stages.0.layers.1.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=64,
             out_channels=64,
             batch_size=8,
@@ -420,7 +485,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_7 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_6,
             weight_tensor=self.weights["resnet.encoder.stages.0.layers.1.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=64,
             out_channels=256,
             batch_size=8,
@@ -460,7 +525,7 @@ class ModelTTNN(LightweightModule):
         )
         ttnn_add_1 = ttnn.add(
             ttnn_conv2d_7,
-            ttnn_relu_0,
+            x,
             dtype=ttnn.DataType.BFLOAT16,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
@@ -478,7 +543,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_conv2d_7, False)
-        ttnn.deallocate(ttnn_relu_0, False)
+        ttnn.deallocate(x, False)
         ttnn_relu_1 = ttnn.relu(
             ttnn_add_1,
             memory_config=ttnn.MemoryConfig(
@@ -497,10 +562,20 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_1, False)
+        return ttnn_relu_1
+
+
+
+class ResNetBottleNeckLayer_S0_L2(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_conv2d_8 = ttnn.conv2d(
-            input_tensor=ttnn_relu_1,
+            input_tensor=x,
             weight_tensor=self.weights["resnet.encoder.stages.0.layers.2.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=64,
             batch_size=8,
@@ -542,7 +617,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_9 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_8,
             weight_tensor=self.weights["resnet.encoder.stages.0.layers.2.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=64,
             out_channels=64,
             batch_size=8,
@@ -584,7 +659,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_10 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_9,
             weight_tensor=self.weights["resnet.encoder.stages.0.layers.2.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=64,
             out_channels=256,
             batch_size=8,
@@ -624,7 +699,7 @@ class ModelTTNN(LightweightModule):
         )
         ttnn_add_2 = ttnn.add(
             ttnn_conv2d_10,
-            ttnn_relu_1,
+            x,
             dtype=ttnn.DataType.BFLOAT16,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
@@ -642,7 +717,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_conv2d_10, False)
-        ttnn.deallocate(ttnn_relu_1, False)
+        ttnn.deallocate(x, False)
         ttnn_relu_2 = ttnn.relu(
             ttnn_add_2,
             memory_config=ttnn.MemoryConfig(
@@ -661,10 +736,20 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_2, False)
+        return ttnn_relu_2
+
+
+
+class ResNetBottleNeckLayer_S1_L0(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_conv2d_11 = ttnn.conv2d(
-            input_tensor=ttnn_relu_2,
+            input_tensor=x,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.0.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=128,
             batch_size=8,
@@ -706,7 +791,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_12 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_11,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.0.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=128,
             out_channels=128,
             batch_size=8,
@@ -748,7 +833,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_13 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_12,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.0.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=128,
             out_channels=512,
             batch_size=8,
@@ -787,9 +872,9 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn_conv2d_14 = ttnn.conv2d(
-            input_tensor=ttnn_relu_2,
+            input_tensor=x,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.0.shortcut.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=512,
             batch_size=8,
@@ -827,7 +912,7 @@ class ModelTTNN(LightweightModule):
                 ),
             ),
         )
-        ttnn.deallocate(ttnn_relu_2, False)
+        ttnn.deallocate(x, False)
         ttnn_add_3 = ttnn.add(
             ttnn_conv2d_13,
             ttnn_conv2d_14,
@@ -867,10 +952,20 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_3, False)
+        return ttnn_relu_3
+
+
+
+class ResNetBottleNeckLayer_S1_L1(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_conv2d_15 = ttnn.conv2d(
-            input_tensor=ttnn_relu_3,
+            input_tensor=x,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.1.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=512,
             out_channels=128,
             batch_size=8,
@@ -912,7 +1007,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_16 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_15,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.1.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=128,
             out_channels=128,
             batch_size=8,
@@ -954,7 +1049,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_17 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_16,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.1.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=128,
             out_channels=512,
             batch_size=8,
@@ -994,7 +1089,7 @@ class ModelTTNN(LightweightModule):
         )
         ttnn_add_4 = ttnn.add(
             ttnn_conv2d_17,
-            ttnn_relu_3,
+            x,
             dtype=ttnn.DataType.BFLOAT16,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
@@ -1012,7 +1107,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_conv2d_17, False)
-        ttnn.deallocate(ttnn_relu_3, False)
+        ttnn.deallocate(x, False)
         ttnn_relu_4 = ttnn.relu(
             ttnn_add_4,
             memory_config=ttnn.MemoryConfig(
@@ -1031,10 +1126,20 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_4, False)
+        return ttnn_relu_4
+
+
+
+class ResNetBottleNeckLayer_S1_L2(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_conv2d_18 = ttnn.conv2d(
-            input_tensor=ttnn_relu_4,
+            input_tensor=x,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.2.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=512,
             out_channels=128,
             batch_size=8,
@@ -1076,7 +1181,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_19 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_18,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.2.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=128,
             out_channels=128,
             batch_size=8,
@@ -1118,7 +1223,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_20 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_19,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.2.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=128,
             out_channels=512,
             batch_size=8,
@@ -1158,7 +1263,7 @@ class ModelTTNN(LightweightModule):
         )
         ttnn_add_5 = ttnn.add(
             ttnn_conv2d_20,
-            ttnn_relu_4,
+            x,
             dtype=ttnn.DataType.BFLOAT16,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
@@ -1176,7 +1281,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_conv2d_20, False)
-        ttnn.deallocate(ttnn_relu_4, False)
+        ttnn.deallocate(x, False)
         ttnn_relu_5 = ttnn.relu(
             ttnn_add_5,
             memory_config=ttnn.MemoryConfig(
@@ -1195,10 +1300,20 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_5, False)
+        return ttnn_relu_5
+
+
+
+class ResNetBottleNeckLayer_S1_L3(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_conv2d_21 = ttnn.conv2d(
-            input_tensor=ttnn_relu_5,
+            input_tensor=x,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.3.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=512,
             out_channels=128,
             batch_size=8,
@@ -1240,7 +1355,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_22 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_21,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.3.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=128,
             out_channels=128,
             batch_size=8,
@@ -1282,7 +1397,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_23 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_22,
             weight_tensor=self.weights["resnet.encoder.stages.1.layers.3.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=128,
             out_channels=512,
             batch_size=8,
@@ -1322,7 +1437,7 @@ class ModelTTNN(LightweightModule):
         )
         ttnn_add_6 = ttnn.add(
             ttnn_conv2d_23,
-            ttnn_relu_5,
+            x,
             dtype=ttnn.DataType.BFLOAT16,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
@@ -1340,7 +1455,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_conv2d_23, False)
-        ttnn.deallocate(ttnn_relu_5, False)
+        ttnn.deallocate(x, False)
         ttnn_relu_6 = ttnn.relu(
             ttnn_add_6,
             memory_config=ttnn.MemoryConfig(
@@ -1359,10 +1474,20 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_6, False)
+        return ttnn_relu_6
+
+
+
+class ResNetBottleNeckLayer_S2_L0(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_conv2d_24 = ttnn.conv2d(
-            input_tensor=ttnn_relu_6,
+            input_tensor=x,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.0.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=512,
             out_channels=256,
             batch_size=8,
@@ -1404,7 +1529,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_25 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_24,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.0.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=256,
             batch_size=8,
@@ -1461,7 +1586,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_26 = ttnn.conv2d(
             input_tensor=ttnn_to_memory_config_1,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.0.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=1024,
             batch_size=8,
@@ -1497,7 +1622,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn_to_memory_config_2 = ttnn.to_memory_config(
-            ttnn_relu_6,
+            x,
             ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
                 ttnn.BufferType.L1,
@@ -1510,11 +1635,11 @@ class ModelTTNN(LightweightModule):
                 ),
             ),
         )
-        ttnn.deallocate(ttnn_relu_6, False)
+        ttnn.deallocate(x, False)
         ttnn_conv2d_27 = ttnn.conv2d(
             input_tensor=ttnn_to_memory_config_2,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.0.shortcut.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=512,
             out_channels=1024,
             batch_size=8,
@@ -1582,8 +1707,18 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_7, False)
+        return ttnn_relu_7
+
+
+
+class ResNetBottleNeckLayer_S2_L1(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_to_memory_config_3 = ttnn.to_memory_config(
-            ttnn_relu_7,
+            x,
             ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
                 ttnn.BufferType.L1,
@@ -1599,7 +1734,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_28 = ttnn.conv2d(
             input_tensor=ttnn_to_memory_config_3,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.1.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=1024,
             out_channels=256,
             batch_size=8,
@@ -1638,7 +1773,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_29 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_28,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.1.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=256,
             batch_size=8,
@@ -1677,7 +1812,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_30 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_29,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.1.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=1024,
             batch_size=8,
@@ -1714,7 +1849,7 @@ class ModelTTNN(LightweightModule):
         )
         ttnn_add_8 = ttnn.add(
             ttnn_conv2d_30,
-            ttnn_relu_7,
+            x,
             dtype=ttnn.DataType.BFLOAT16,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
@@ -1729,7 +1864,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_conv2d_30, False)
-        ttnn.deallocate(ttnn_relu_7, False)
+        ttnn.deallocate(x, False)
         ttnn_relu_8 = ttnn.relu(
             ttnn_add_8,
             memory_config=ttnn.MemoryConfig(
@@ -1745,8 +1880,18 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_8, False)
+        return ttnn_relu_8
+
+
+
+class ResNetBottleNeckLayer_S2_L2(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_to_memory_config_4 = ttnn.to_memory_config(
-            ttnn_relu_8,
+            x,
             ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
                 ttnn.BufferType.L1,
@@ -1762,7 +1907,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_31 = ttnn.conv2d(
             input_tensor=ttnn_to_memory_config_4,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.2.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=1024,
             out_channels=256,
             batch_size=8,
@@ -1801,7 +1946,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_32 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_31,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.2.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=256,
             batch_size=8,
@@ -1840,7 +1985,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_33 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_32,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.2.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=1024,
             batch_size=8,
@@ -1877,7 +2022,7 @@ class ModelTTNN(LightweightModule):
         )
         ttnn_add_9 = ttnn.add(
             ttnn_conv2d_33,
-            ttnn_relu_8,
+            x,
             dtype=ttnn.DataType.BFLOAT16,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
@@ -1892,7 +2037,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_conv2d_33, False)
-        ttnn.deallocate(ttnn_relu_8, False)
+        ttnn.deallocate(x, False)
         ttnn_relu_9 = ttnn.relu(
             ttnn_add_9,
             memory_config=ttnn.MemoryConfig(
@@ -1908,8 +2053,18 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_9, False)
+        return ttnn_relu_9
+
+
+
+class ResNetBottleNeckLayer_S2_L3(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_to_memory_config_5 = ttnn.to_memory_config(
-            ttnn_relu_9,
+            x,
             ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
                 ttnn.BufferType.L1,
@@ -1925,7 +2080,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_34 = ttnn.conv2d(
             input_tensor=ttnn_to_memory_config_5,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.3.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=1024,
             out_channels=256,
             batch_size=8,
@@ -1964,7 +2119,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_35 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_34,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.3.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=256,
             batch_size=8,
@@ -2003,7 +2158,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_36 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_35,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.3.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=1024,
             batch_size=8,
@@ -2040,7 +2195,7 @@ class ModelTTNN(LightweightModule):
         )
         ttnn_add_10 = ttnn.add(
             ttnn_conv2d_36,
-            ttnn_relu_9,
+            x,
             dtype=ttnn.DataType.BFLOAT16,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
@@ -2055,7 +2210,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_conv2d_36, False)
-        ttnn.deallocate(ttnn_relu_9, False)
+        ttnn.deallocate(x, False)
         ttnn_relu_10 = ttnn.relu(
             ttnn_add_10,
             memory_config=ttnn.MemoryConfig(
@@ -2071,8 +2226,18 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_10, False)
+        return ttnn_relu_10
+
+
+
+class ResNetBottleNeckLayer_S2_L4(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_to_memory_config_6 = ttnn.to_memory_config(
-            ttnn_relu_10,
+            x,
             ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
                 ttnn.BufferType.L1,
@@ -2088,7 +2253,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_37 = ttnn.conv2d(
             input_tensor=ttnn_to_memory_config_6,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.4.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=1024,
             out_channels=256,
             batch_size=8,
@@ -2127,7 +2292,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_38 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_37,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.4.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=256,
             batch_size=8,
@@ -2166,7 +2331,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_39 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_38,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.4.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=1024,
             batch_size=8,
@@ -2203,7 +2368,7 @@ class ModelTTNN(LightweightModule):
         )
         ttnn_add_11 = ttnn.add(
             ttnn_conv2d_39,
-            ttnn_relu_10,
+            x,
             dtype=ttnn.DataType.BFLOAT16,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
@@ -2218,7 +2383,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_conv2d_39, False)
-        ttnn.deallocate(ttnn_relu_10, False)
+        ttnn.deallocate(x, False)
         ttnn_relu_11 = ttnn.relu(
             ttnn_add_11,
             memory_config=ttnn.MemoryConfig(
@@ -2234,8 +2399,18 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_11, False)
+        return ttnn_relu_11
+
+
+
+class ResNetBottleNeckLayer_S2_L5(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_to_memory_config_7 = ttnn.to_memory_config(
-            ttnn_relu_11,
+            x,
             ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
                 ttnn.BufferType.L1,
@@ -2251,7 +2426,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_40 = ttnn.conv2d(
             input_tensor=ttnn_to_memory_config_7,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.5.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=1024,
             out_channels=256,
             batch_size=8,
@@ -2290,7 +2465,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_41 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_40,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.5.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=256,
             batch_size=8,
@@ -2329,7 +2504,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_42 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_41,
             weight_tensor=self.weights["resnet.encoder.stages.2.layers.5.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=256,
             out_channels=1024,
             batch_size=8,
@@ -2366,7 +2541,7 @@ class ModelTTNN(LightweightModule):
         )
         ttnn_add_12 = ttnn.add(
             ttnn_conv2d_42,
-            ttnn_relu_11,
+            x,
             dtype=ttnn.DataType.BFLOAT16,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
@@ -2381,7 +2556,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_conv2d_42, False)
-        ttnn.deallocate(ttnn_relu_11, False)
+        ttnn.deallocate(x, False)
         ttnn_relu_12 = ttnn.relu(
             ttnn_add_12,
             memory_config=ttnn.MemoryConfig(
@@ -2397,8 +2572,18 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_12, False)
+        return ttnn_relu_12
+
+
+
+class ResNetBottleNeckLayer_S3_L0(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_to_memory_config_8 = ttnn.to_memory_config(
-            ttnn_relu_12,
+            x,
             ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
                 ttnn.BufferType.L1,
@@ -2414,7 +2599,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_43 = ttnn.conv2d(
             input_tensor=ttnn_to_memory_config_8,
             weight_tensor=self.weights["resnet.encoder.stages.3.layers.0.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=1024,
             out_channels=512,
             batch_size=8,
@@ -2453,7 +2638,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_44 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_43,
             weight_tensor=self.weights["resnet.encoder.stages.3.layers.0.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=512,
             out_channels=512,
             batch_size=8,
@@ -2492,7 +2677,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_45 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_44,
             weight_tensor=self.weights["resnet.encoder.stages.3.layers.0.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=512,
             out_channels=2048,
             batch_size=8,
@@ -2528,9 +2713,9 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn_conv2d_46 = ttnn.conv2d(
-            input_tensor=ttnn_relu_12,
+            input_tensor=x,
             weight_tensor=self.weights["resnet.encoder.stages.3.layers.0.shortcut.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=1024,
             out_channels=2048,
             batch_size=8,
@@ -2565,7 +2750,7 @@ class ModelTTNN(LightweightModule):
                 ),
             ),
         )
-        ttnn.deallocate(ttnn_relu_12, False)
+        ttnn.deallocate(x, False)
         ttnn_add_13 = ttnn.add(
             ttnn_conv2d_45,
             ttnn_conv2d_46,
@@ -2599,8 +2784,18 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_13, False)
+        return ttnn_relu_13
+
+
+
+class ResNetBottleNeckLayer_S3_L1(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_to_memory_config_9 = ttnn.to_memory_config(
-            ttnn_relu_13,
+            x,
             ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
                 ttnn.BufferType.L1,
@@ -2616,7 +2811,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_47 = ttnn.conv2d(
             input_tensor=ttnn_to_memory_config_9,
             weight_tensor=self.weights["resnet.encoder.stages.3.layers.1.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=2048,
             out_channels=512,
             batch_size=8,
@@ -2655,7 +2850,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_48 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_47,
             weight_tensor=self.weights["resnet.encoder.stages.3.layers.1.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=512,
             out_channels=512,
             batch_size=8,
@@ -2694,7 +2889,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_49 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_48,
             weight_tensor=self.weights["resnet.encoder.stages.3.layers.1.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=512,
             out_channels=2048,
             batch_size=8,
@@ -2731,7 +2926,7 @@ class ModelTTNN(LightweightModule):
         )
         ttnn_add_14 = ttnn.add(
             ttnn_conv2d_49,
-            ttnn_relu_13,
+            x,
             dtype=ttnn.DataType.BFLOAT16,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
@@ -2746,7 +2941,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_conv2d_49, False)
-        ttnn.deallocate(ttnn_relu_13, False)
+        ttnn.deallocate(x, False)
         ttnn_relu_14 = ttnn.relu(
             ttnn_add_14,
             memory_config=ttnn.MemoryConfig(
@@ -2762,8 +2957,18 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_14, False)
+        return ttnn_relu_14
+
+
+
+class ResNetBottleNeckLayer_S3_L2(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_to_memory_config_10 = ttnn.to_memory_config(
-            ttnn_relu_14,
+            x,
             ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
                 ttnn.BufferType.L1,
@@ -2779,7 +2984,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_50 = ttnn.conv2d(
             input_tensor=ttnn_to_memory_config_10,
             weight_tensor=self.weights["resnet.encoder.stages.3.layers.2.layer.0.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=2048,
             out_channels=512,
             batch_size=8,
@@ -2818,7 +3023,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_51 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_50,
             weight_tensor=self.weights["resnet.encoder.stages.3.layers.2.layer.1.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=512,
             out_channels=512,
             batch_size=8,
@@ -2857,7 +3062,7 @@ class ModelTTNN(LightweightModule):
         ttnn_conv2d_52 = ttnn.conv2d(
             input_tensor=ttnn_conv2d_51,
             weight_tensor=self.weights["resnet.encoder.stages.3.layers.2.layer.2.folded_conv_weight"],
-            device=utils_DeviceGetter_get_device_0,
+            device=self.device,
             in_channels=512,
             out_channels=2048,
             batch_size=8,
@@ -2894,7 +3099,7 @@ class ModelTTNN(LightweightModule):
         )
         ttnn_add_15 = ttnn.add(
             ttnn_conv2d_52,
-            ttnn_relu_14,
+            x,
             dtype=ttnn.DataType.BFLOAT16,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
@@ -2909,7 +3114,7 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_conv2d_52, False)
-        ttnn.deallocate(ttnn_relu_14, False)
+        ttnn.deallocate(x, False)
         ttnn_relu_15 = ttnn.relu(
             ttnn_add_15,
             memory_config=ttnn.MemoryConfig(
@@ -2925,8 +3130,18 @@ class ModelTTNN(LightweightModule):
             ),
         )
         ttnn.deallocate(ttnn_add_15, False)
+        return ttnn_relu_15
+
+
+
+class ResNetClassifier(LightweightModule):
+    def __init__(self, device, weights):
+        self.device = device
+        self.weights = weights
+
+    def forward(self, x):
         ttnn_typecast_0 = ttnn.typecast(
-            ttnn_relu_15,
+            x,
             ttnn.DataType.FLOAT32,
             memory_config=ttnn.MemoryConfig(
                 ttnn.TensorMemoryLayout.BLOCK_SHARDED,
@@ -2940,7 +3155,7 @@ class ModelTTNN(LightweightModule):
                 ),
             ),
         )
-        ttnn.deallocate(ttnn_relu_15, False)
+        ttnn.deallocate(x, False)
         ttnn_reshape_1 = ttnn.reshape(
             ttnn_typecast_0,
             [8, 1, 49, 2048],
@@ -3042,3 +3257,4 @@ class ModelTTNN(LightweightModule):
         )
         ttnn.deallocate(ttnn_typecast_1, False)
         return [ttnn_to_memory_config_11]
+
