diff --git a/Qwen/Qwen3-4B/model/graph_0/consteval.py b/Qwen/Qwen3-4B/model/graph_0/consteval.py
index cfefa49..6f409c7 100644
--- a/Qwen/Qwen3-4B/model/graph_0/consteval.py
+++ b/Qwen/Qwen3-4B/model/graph_0/consteval.py
@@ -1,14 +1,12 @@
 import ttnn
-import utils
 import ttir_cpu
 import torch
 
 
-def main_const_eval_0(arg):
-    utils_DeviceGetter_get_device_2 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_0(arg, device):
     ttnn_to_device_0 = ttnn.to_device(
         arg[0],
-        device=utils_DeviceGetter_get_device_2,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -29,8 +27,7 @@ def cpu_hoisted_const_eval_70989bca(arg_0, arg_1, arg_2):
     return ttnn_from_torch_1
 
 
-def main_const_eval_1(arg):
-    utils_DeviceGetter_get_device_3 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_1(arg, device):
     ttnn_typecast_1 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_2 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_3 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -50,7 +47,7 @@ def main_const_eval_1(arg):
     ttnn.deallocate(ttnn_typecast_4, False)
     ttnn_to_device_1 = ttnn.to_device(
         ttnn_to_layout_0,
-        device=utils_DeviceGetter_get_device_3,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -79,8 +76,7 @@ def cpu_hoisted_const_eval_d18be0b6(arg):
     return ttnn_from_torch_2, ttnn_from_torch_3
 
 
-def main_const_eval_2(arg):
-    utils_DeviceGetter_get_device_4 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_2(arg, device):
     v_0, v_1 = cpu_hoisted_const_eval_d18be0b6(arg[0])
     ttnn_typecast_5 = ttnn.typecast(v_0, ttnn.DataType.BFLOAT16, memory_config=None)
     ttnn.deallocate(v_0, False)
@@ -90,7 +86,7 @@ def main_const_eval_2(arg):
     ttnn.deallocate(ttnn_typecast_5, False)
     ttnn_to_device_2 = ttnn.to_device(
         ttnn_to_layout_1,
-        device=utils_DeviceGetter_get_device_4,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -104,7 +100,7 @@ def main_const_eval_2(arg):
     ttnn.deallocate(ttnn_typecast_6, False)
     ttnn_to_device_3 = ttnn.to_device(
         ttnn_to_layout_2,
-        device=utils_DeviceGetter_get_device_4,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -113,8 +109,7 @@ def main_const_eval_2(arg):
     return [ttnn_to_device_2, ttnn_to_device_3]
 
 
-def main_const_eval_3(arg):
-    utils_DeviceGetter_get_device_5 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_3(arg, device):
     ttnn_typecast_7 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_8 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_9 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -134,7 +129,7 @@ def main_const_eval_3(arg):
     ttnn.deallocate(ttnn_typecast_10, False)
     ttnn_to_device_4 = ttnn.to_device(
         ttnn_to_layout_3,
-        device=utils_DeviceGetter_get_device_5,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -143,8 +138,7 @@ def main_const_eval_3(arg):
     return [ttnn_to_device_4]
 
 
-def main_const_eval_4(arg):
-    utils_DeviceGetter_get_device_6 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_4(arg, device):
     ttnn_typecast_11 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_12 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_13 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -164,7 +158,7 @@ def main_const_eval_4(arg):
     ttnn.deallocate(ttnn_typecast_14, False)
     ttnn_to_device_5 = ttnn.to_device(
         ttnn_to_layout_4,
-        device=utils_DeviceGetter_get_device_6,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -173,8 +167,7 @@ def main_const_eval_4(arg):
     return [ttnn_to_device_5]
 
 
-def main_const_eval_5(arg):
-    utils_DeviceGetter_get_device_7 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_5(arg, device):
     ttnn_typecast_15 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_16 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_17 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -194,7 +187,7 @@ def main_const_eval_5(arg):
     ttnn.deallocate(ttnn_typecast_18, False)
     ttnn_to_device_6 = ttnn.to_device(
         ttnn_to_layout_5,
-        device=utils_DeviceGetter_get_device_7,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -203,8 +196,7 @@ def main_const_eval_5(arg):
     return [ttnn_to_device_6]
 
 
-def main_const_eval_6(arg):
-    utils_DeviceGetter_get_device_8 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_6(arg, device):
     ttnn_typecast_19 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_20 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_21 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -224,7 +216,7 @@ def main_const_eval_6(arg):
     ttnn.deallocate(ttnn_typecast_22, False)
     ttnn_to_device_7 = ttnn.to_device(
         ttnn_to_layout_6,
-        device=utils_DeviceGetter_get_device_8,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -233,8 +225,7 @@ def main_const_eval_6(arg):
     return [ttnn_to_device_7]
 
 
-def main_const_eval_7(arg):
-    utils_DeviceGetter_get_device_9 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_7(arg, device):
     ttnn_typecast_23 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_24 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_25 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -254,7 +245,7 @@ def main_const_eval_7(arg):
     ttnn.deallocate(ttnn_typecast_26, False)
     ttnn_to_device_8 = ttnn.to_device(
         ttnn_to_layout_7,
-        device=utils_DeviceGetter_get_device_9,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -263,8 +254,7 @@ def main_const_eval_7(arg):
     return [ttnn_to_device_8]
 
 
-def main_const_eval_8(arg):
-    utils_DeviceGetter_get_device_10 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_8(arg, device):
     ttnn_typecast_27 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_28 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_29 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -284,7 +274,7 @@ def main_const_eval_8(arg):
     ttnn.deallocate(ttnn_typecast_30, False)
     ttnn_to_device_9 = ttnn.to_device(
         ttnn_to_layout_8,
-        device=utils_DeviceGetter_get_device_10,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -293,8 +283,7 @@ def main_const_eval_8(arg):
     return [ttnn_to_device_9]
 
 
-def main_const_eval_9(arg):
-    utils_DeviceGetter_get_device_11 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_9(arg, device):
     ttnn_typecast_31 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_32 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_33 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -314,7 +303,7 @@ def main_const_eval_9(arg):
     ttnn.deallocate(ttnn_typecast_34, False)
     ttnn_to_device_10 = ttnn.to_device(
         ttnn_to_layout_9,
-        device=utils_DeviceGetter_get_device_11,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -323,8 +312,7 @@ def main_const_eval_9(arg):
     return [ttnn_to_device_10]
 
 
-def main_const_eval_10(arg):
-    utils_DeviceGetter_get_device_12 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_10(arg, device):
     ttnn_typecast_35 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_36 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_37 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -344,7 +332,7 @@ def main_const_eval_10(arg):
     ttnn.deallocate(ttnn_typecast_38, False)
     ttnn_to_device_11 = ttnn.to_device(
         ttnn_to_layout_10,
-        device=utils_DeviceGetter_get_device_12,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -353,8 +341,7 @@ def main_const_eval_10(arg):
     return [ttnn_to_device_11]
 
 
-def main_const_eval_11(arg):
-    utils_DeviceGetter_get_device_13 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_11(arg, device):
     ttnn_typecast_39 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_40 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_41 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -374,7 +361,7 @@ def main_const_eval_11(arg):
     ttnn.deallocate(ttnn_typecast_42, False)
     ttnn_to_device_12 = ttnn.to_device(
         ttnn_to_layout_11,
-        device=utils_DeviceGetter_get_device_13,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -383,8 +370,7 @@ def main_const_eval_11(arg):
     return [ttnn_to_device_12]
 
 
-def main_const_eval_12(arg):
-    utils_DeviceGetter_get_device_14 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_12(arg, device):
     ttnn_typecast_43 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_44 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_45 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -404,7 +390,7 @@ def main_const_eval_12(arg):
     ttnn.deallocate(ttnn_typecast_46, False)
     ttnn_to_device_13 = ttnn.to_device(
         ttnn_to_layout_12,
-        device=utils_DeviceGetter_get_device_14,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -413,8 +399,7 @@ def main_const_eval_12(arg):
     return [ttnn_to_device_13]
 
 
-def main_const_eval_13(arg):
-    utils_DeviceGetter_get_device_15 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_13(arg, device):
     ttnn_typecast_47 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_48 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_49 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -434,7 +419,7 @@ def main_const_eval_13(arg):
     ttnn.deallocate(ttnn_typecast_50, False)
     ttnn_to_device_14 = ttnn.to_device(
         ttnn_to_layout_13,
-        device=utils_DeviceGetter_get_device_15,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -443,8 +428,7 @@ def main_const_eval_13(arg):
     return [ttnn_to_device_14]
 
 
-def main_const_eval_14(arg):
-    utils_DeviceGetter_get_device_16 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_14(arg, device):
     ttnn_typecast_51 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_52 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_53 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -464,7 +448,7 @@ def main_const_eval_14(arg):
     ttnn.deallocate(ttnn_typecast_54, False)
     ttnn_to_device_15 = ttnn.to_device(
         ttnn_to_layout_14,
-        device=utils_DeviceGetter_get_device_16,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -473,8 +457,7 @@ def main_const_eval_14(arg):
     return [ttnn_to_device_15]
 
 
-def main_const_eval_15(arg):
-    utils_DeviceGetter_get_device_17 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_15(arg, device):
     ttnn_typecast_55 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_56 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_57 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -494,7 +477,7 @@ def main_const_eval_15(arg):
     ttnn.deallocate(ttnn_typecast_58, False)
     ttnn_to_device_16 = ttnn.to_device(
         ttnn_to_layout_15,
-        device=utils_DeviceGetter_get_device_17,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -503,8 +486,7 @@ def main_const_eval_15(arg):
     return [ttnn_to_device_16]
 
 
-def main_const_eval_16(arg):
-    utils_DeviceGetter_get_device_18 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_16(arg, device):
     ttnn_typecast_59 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_60 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_61 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -524,7 +506,7 @@ def main_const_eval_16(arg):
     ttnn.deallocate(ttnn_typecast_62, False)
     ttnn_to_device_17 = ttnn.to_device(
         ttnn_to_layout_16,
-        device=utils_DeviceGetter_get_device_18,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -533,8 +515,7 @@ def main_const_eval_16(arg):
     return [ttnn_to_device_17]
 
 
-def main_const_eval_17(arg):
-    utils_DeviceGetter_get_device_19 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_17(arg, device):
     ttnn_typecast_63 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_64 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_65 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -554,7 +535,7 @@ def main_const_eval_17(arg):
     ttnn.deallocate(ttnn_typecast_66, False)
     ttnn_to_device_18 = ttnn.to_device(
         ttnn_to_layout_17,
-        device=utils_DeviceGetter_get_device_19,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -563,8 +544,7 @@ def main_const_eval_17(arg):
     return [ttnn_to_device_18]
 
 
-def main_const_eval_18(arg):
-    utils_DeviceGetter_get_device_20 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_18(arg, device):
     ttnn_typecast_67 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_68 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_69 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -584,7 +564,7 @@ def main_const_eval_18(arg):
     ttnn.deallocate(ttnn_typecast_70, False)
     ttnn_to_device_19 = ttnn.to_device(
         ttnn_to_layout_18,
-        device=utils_DeviceGetter_get_device_20,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -593,8 +573,7 @@ def main_const_eval_18(arg):
     return [ttnn_to_device_19]
 
 
-def main_const_eval_19(arg):
-    utils_DeviceGetter_get_device_21 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_19(arg, device):
     ttnn_typecast_71 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_72 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_73 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -614,7 +593,7 @@ def main_const_eval_19(arg):
     ttnn.deallocate(ttnn_typecast_74, False)
     ttnn_to_device_20 = ttnn.to_device(
         ttnn_to_layout_19,
-        device=utils_DeviceGetter_get_device_21,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -623,8 +602,7 @@ def main_const_eval_19(arg):
     return [ttnn_to_device_20]
 
 
-def main_const_eval_20(arg):
-    utils_DeviceGetter_get_device_22 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_20(arg, device):
     ttnn_typecast_75 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_76 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_77 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -644,7 +622,7 @@ def main_const_eval_20(arg):
     ttnn.deallocate(ttnn_typecast_78, False)
     ttnn_to_device_21 = ttnn.to_device(
         ttnn_to_layout_20,
-        device=utils_DeviceGetter_get_device_22,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -676,8 +654,7 @@ def cpu_hoisted_const_eval_17957e90():
     return ttnn_from_torch_0
 
 
-def main_const_eval_21():
-    utils_DeviceGetter_get_device_23 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_21(device):
     cpu_hoisted_const_eval_17957e90_0 = cpu_hoisted_const_eval_17957e90()
     ttnn_typecast_79 = ttnn.typecast(
         cpu_hoisted_const_eval_17957e90_0, ttnn.DataType.BFLOAT16, memory_config=None
@@ -688,7 +665,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_79, False)
     ttnn_to_device_22 = ttnn.to_device(
         ttnn_to_layout_21,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -703,7 +680,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_80, False)
     ttnn_to_device_23 = ttnn.to_device(
         ttnn_to_layout_22,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -718,7 +695,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_81, False)
     ttnn_to_device_24 = ttnn.to_device(
         ttnn_to_layout_23,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -733,7 +710,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_82, False)
     ttnn_to_device_25 = ttnn.to_device(
         ttnn_to_layout_24,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -748,7 +725,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_83, False)
     ttnn_to_device_26 = ttnn.to_device(
         ttnn_to_layout_25,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -763,7 +740,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_84, False)
     ttnn_to_device_27 = ttnn.to_device(
         ttnn_to_layout_26,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -778,7 +755,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_85, False)
     ttnn_to_device_28 = ttnn.to_device(
         ttnn_to_layout_27,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -793,7 +770,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_86, False)
     ttnn_to_device_29 = ttnn.to_device(
         ttnn_to_layout_28,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -808,7 +785,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_87, False)
     ttnn_to_device_30 = ttnn.to_device(
         ttnn_to_layout_29,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -823,7 +800,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_88, False)
     ttnn_to_device_31 = ttnn.to_device(
         ttnn_to_layout_30,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -838,7 +815,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_89, False)
     ttnn_to_device_32 = ttnn.to_device(
         ttnn_to_layout_31,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -853,7 +830,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_90, False)
     ttnn_to_device_33 = ttnn.to_device(
         ttnn_to_layout_32,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -868,7 +845,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_91, False)
     ttnn_to_device_34 = ttnn.to_device(
         ttnn_to_layout_33,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -883,7 +860,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_92, False)
     ttnn_to_device_35 = ttnn.to_device(
         ttnn_to_layout_34,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -898,7 +875,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_93, False)
     ttnn_to_device_36 = ttnn.to_device(
         ttnn_to_layout_35,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -913,7 +890,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_94, False)
     ttnn_to_device_37 = ttnn.to_device(
         ttnn_to_layout_36,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -928,7 +905,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_95, False)
     ttnn_to_device_38 = ttnn.to_device(
         ttnn_to_layout_37,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -943,7 +920,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_96, False)
     ttnn_to_device_39 = ttnn.to_device(
         ttnn_to_layout_38,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -958,7 +935,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_97, False)
     ttnn_to_device_40 = ttnn.to_device(
         ttnn_to_layout_39,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -973,7 +950,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_98, False)
     ttnn_to_device_41 = ttnn.to_device(
         ttnn_to_layout_40,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -988,7 +965,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_99, False)
     ttnn_to_device_42 = ttnn.to_device(
         ttnn_to_layout_41,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1003,7 +980,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_100, False)
     ttnn_to_device_43 = ttnn.to_device(
         ttnn_to_layout_42,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1018,7 +995,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_101, False)
     ttnn_to_device_44 = ttnn.to_device(
         ttnn_to_layout_43,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1033,7 +1010,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_102, False)
     ttnn_to_device_45 = ttnn.to_device(
         ttnn_to_layout_44,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1048,7 +1025,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_103, False)
     ttnn_to_device_46 = ttnn.to_device(
         ttnn_to_layout_45,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1063,7 +1040,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_104, False)
     ttnn_to_device_47 = ttnn.to_device(
         ttnn_to_layout_46,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1078,7 +1055,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_105, False)
     ttnn_to_device_48 = ttnn.to_device(
         ttnn_to_layout_47,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1093,7 +1070,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_106, False)
     ttnn_to_device_49 = ttnn.to_device(
         ttnn_to_layout_48,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1108,7 +1085,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_107, False)
     ttnn_to_device_50 = ttnn.to_device(
         ttnn_to_layout_49,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1123,7 +1100,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_108, False)
     ttnn_to_device_51 = ttnn.to_device(
         ttnn_to_layout_50,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1138,7 +1115,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_109, False)
     ttnn_to_device_52 = ttnn.to_device(
         ttnn_to_layout_51,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1153,7 +1130,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_110, False)
     ttnn_to_device_53 = ttnn.to_device(
         ttnn_to_layout_52,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1168,7 +1145,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_111, False)
     ttnn_to_device_54 = ttnn.to_device(
         ttnn_to_layout_53,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1183,7 +1160,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_112, False)
     ttnn_to_device_55 = ttnn.to_device(
         ttnn_to_layout_54,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1198,7 +1175,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_113, False)
     ttnn_to_device_56 = ttnn.to_device(
         ttnn_to_layout_55,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1214,7 +1191,7 @@ def main_const_eval_21():
     ttnn.deallocate(ttnn_typecast_114, False)
     ttnn_to_device_57 = ttnn.to_device(
         ttnn_to_layout_56,
-        device=utils_DeviceGetter_get_device_23,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1260,8 +1237,7 @@ def main_const_eval_21():
     ]
 
 
-def main_const_eval_22(arg):
-    utils_DeviceGetter_get_device_24 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_22(arg, device):
     ttnn_typecast_115 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_116 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_117 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1281,7 +1257,7 @@ def main_const_eval_22(arg):
     ttnn.deallocate(ttnn_typecast_118, False)
     ttnn_to_device_58 = ttnn.to_device(
         ttnn_to_layout_57,
-        device=utils_DeviceGetter_get_device_24,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1290,8 +1266,7 @@ def main_const_eval_22(arg):
     return [ttnn_to_device_58]
 
 
-def main_const_eval_23(arg):
-    utils_DeviceGetter_get_device_25 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_23(arg, device):
     ttnn_typecast_119 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_120 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_121 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1311,7 +1286,7 @@ def main_const_eval_23(arg):
     ttnn.deallocate(ttnn_typecast_122, False)
     ttnn_to_device_59 = ttnn.to_device(
         ttnn_to_layout_58,
-        device=utils_DeviceGetter_get_device_25,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1320,8 +1295,7 @@ def main_const_eval_23(arg):
     return [ttnn_to_device_59]
 
 
-def main_const_eval_24(arg):
-    utils_DeviceGetter_get_device_26 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_24(arg, device):
     ttnn_typecast_123 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_124 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_125 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1341,7 +1315,7 @@ def main_const_eval_24(arg):
     ttnn.deallocate(ttnn_typecast_126, False)
     ttnn_to_device_60 = ttnn.to_device(
         ttnn_to_layout_59,
-        device=utils_DeviceGetter_get_device_26,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1350,8 +1324,7 @@ def main_const_eval_24(arg):
     return [ttnn_to_device_60]
 
 
-def main_const_eval_25(arg):
-    utils_DeviceGetter_get_device_27 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_25(arg, device):
     ttnn_typecast_127 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_128 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_129 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1371,7 +1344,7 @@ def main_const_eval_25(arg):
     ttnn.deallocate(ttnn_typecast_130, False)
     ttnn_to_device_61 = ttnn.to_device(
         ttnn_to_layout_60,
-        device=utils_DeviceGetter_get_device_27,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1380,8 +1353,7 @@ def main_const_eval_25(arg):
     return [ttnn_to_device_61]
 
 
-def main_const_eval_26(arg):
-    utils_DeviceGetter_get_device_28 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_26(arg, device):
     ttnn_typecast_131 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_132 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_133 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1401,7 +1373,7 @@ def main_const_eval_26(arg):
     ttnn.deallocate(ttnn_typecast_134, False)
     ttnn_to_device_62 = ttnn.to_device(
         ttnn_to_layout_61,
-        device=utils_DeviceGetter_get_device_28,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1410,8 +1382,7 @@ def main_const_eval_26(arg):
     return [ttnn_to_device_62]
 
 
-def main_const_eval_27(arg):
-    utils_DeviceGetter_get_device_29 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_27(arg, device):
     ttnn_typecast_135 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_136 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_137 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1431,7 +1402,7 @@ def main_const_eval_27(arg):
     ttnn.deallocate(ttnn_typecast_138, False)
     ttnn_to_device_63 = ttnn.to_device(
         ttnn_to_layout_62,
-        device=utils_DeviceGetter_get_device_29,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1440,8 +1411,7 @@ def main_const_eval_27(arg):
     return [ttnn_to_device_63]
 
 
-def main_const_eval_28(arg):
-    utils_DeviceGetter_get_device_30 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_28(arg, device):
     ttnn_typecast_139 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_140 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_141 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1461,7 +1431,7 @@ def main_const_eval_28(arg):
     ttnn.deallocate(ttnn_typecast_142, False)
     ttnn_to_device_64 = ttnn.to_device(
         ttnn_to_layout_63,
-        device=utils_DeviceGetter_get_device_30,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1470,8 +1440,7 @@ def main_const_eval_28(arg):
     return [ttnn_to_device_64]
 
 
-def main_const_eval_29(arg):
-    utils_DeviceGetter_get_device_31 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_29(arg, device):
     ttnn_typecast_143 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_144 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_145 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1491,7 +1460,7 @@ def main_const_eval_29(arg):
     ttnn.deallocate(ttnn_typecast_146, False)
     ttnn_to_device_65 = ttnn.to_device(
         ttnn_to_layout_64,
-        device=utils_DeviceGetter_get_device_31,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1500,8 +1469,7 @@ def main_const_eval_29(arg):
     return [ttnn_to_device_65]
 
 
-def main_const_eval_30(arg):
-    utils_DeviceGetter_get_device_32 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_30(arg, device):
     ttnn_typecast_147 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_148 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_149 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1521,7 +1489,7 @@ def main_const_eval_30(arg):
     ttnn.deallocate(ttnn_typecast_150, False)
     ttnn_to_device_66 = ttnn.to_device(
         ttnn_to_layout_65,
-        device=utils_DeviceGetter_get_device_32,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1530,8 +1498,7 @@ def main_const_eval_30(arg):
     return [ttnn_to_device_66]
 
 
-def main_const_eval_31(arg):
-    utils_DeviceGetter_get_device_33 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_31(arg, device):
     ttnn_typecast_151 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_152 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_153 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1551,7 +1518,7 @@ def main_const_eval_31(arg):
     ttnn.deallocate(ttnn_typecast_154, False)
     ttnn_to_device_67 = ttnn.to_device(
         ttnn_to_layout_66,
-        device=utils_DeviceGetter_get_device_33,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1560,8 +1527,7 @@ def main_const_eval_31(arg):
     return [ttnn_to_device_67]
 
 
-def main_const_eval_32(arg):
-    utils_DeviceGetter_get_device_34 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_32(arg, device):
     ttnn_typecast_155 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_156 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_157 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1581,7 +1547,7 @@ def main_const_eval_32(arg):
     ttnn.deallocate(ttnn_typecast_158, False)
     ttnn_to_device_68 = ttnn.to_device(
         ttnn_to_layout_67,
-        device=utils_DeviceGetter_get_device_34,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1590,8 +1556,7 @@ def main_const_eval_32(arg):
     return [ttnn_to_device_68]
 
 
-def main_const_eval_33(arg):
-    utils_DeviceGetter_get_device_35 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_33(arg, device):
     ttnn_typecast_159 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_160 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_161 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1611,7 +1576,7 @@ def main_const_eval_33(arg):
     ttnn.deallocate(ttnn_typecast_162, False)
     ttnn_to_device_69 = ttnn.to_device(
         ttnn_to_layout_68,
-        device=utils_DeviceGetter_get_device_35,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1620,8 +1585,7 @@ def main_const_eval_33(arg):
     return [ttnn_to_device_69]
 
 
-def main_const_eval_34(arg):
-    utils_DeviceGetter_get_device_36 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_34(arg, device):
     ttnn_typecast_163 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_164 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_165 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1641,7 +1605,7 @@ def main_const_eval_34(arg):
     ttnn.deallocate(ttnn_typecast_166, False)
     ttnn_to_device_70 = ttnn.to_device(
         ttnn_to_layout_69,
-        device=utils_DeviceGetter_get_device_36,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1650,8 +1614,7 @@ def main_const_eval_34(arg):
     return [ttnn_to_device_70]
 
 
-def main_const_eval_35(arg):
-    utils_DeviceGetter_get_device_37 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_35(arg, device):
     ttnn_typecast_167 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_168 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_169 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1671,7 +1634,7 @@ def main_const_eval_35(arg):
     ttnn.deallocate(ttnn_typecast_170, False)
     ttnn_to_device_71 = ttnn.to_device(
         ttnn_to_layout_70,
-        device=utils_DeviceGetter_get_device_37,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1680,8 +1643,7 @@ def main_const_eval_35(arg):
     return [ttnn_to_device_71]
 
 
-def main_const_eval_36(arg):
-    utils_DeviceGetter_get_device_38 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_36(arg, device):
     ttnn_typecast_171 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_172 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_173 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1701,7 +1663,7 @@ def main_const_eval_36(arg):
     ttnn.deallocate(ttnn_typecast_174, False)
     ttnn_to_device_72 = ttnn.to_device(
         ttnn_to_layout_71,
-        device=utils_DeviceGetter_get_device_38,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1710,8 +1672,7 @@ def main_const_eval_36(arg):
     return [ttnn_to_device_72]
 
 
-def main_const_eval_37(arg):
-    utils_DeviceGetter_get_device_39 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_37(arg, device):
     ttnn_typecast_175 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_176 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_177 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1731,7 +1692,7 @@ def main_const_eval_37(arg):
     ttnn.deallocate(ttnn_typecast_178, False)
     ttnn_to_device_73 = ttnn.to_device(
         ttnn_to_layout_72,
-        device=utils_DeviceGetter_get_device_39,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1740,8 +1701,7 @@ def main_const_eval_37(arg):
     return [ttnn_to_device_73]
 
 
-def main_const_eval_38(arg):
-    utils_DeviceGetter_get_device_40 = utils.DeviceGetter.get_device((1, 1))
+def main_const_eval_38(arg, device):
     ttnn_typecast_179 = ttnn.typecast(arg[2], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_180 = ttnn.typecast(arg[1], ttnn.DataType.FLOAT32, memory_config=None)
     ttnn_typecast_181 = ttnn.typecast(arg[0], ttnn.DataType.FLOAT32, memory_config=None)
@@ -1761,7 +1721,7 @@ def main_const_eval_38(arg):
     ttnn.deallocate(ttnn_typecast_182, False)
     ttnn_to_device_74 = ttnn.to_device(
         ttnn_to_layout_73,
-        device=utils_DeviceGetter_get_device_40,
+        device=device,
         memory_config=ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -1770,10 +1730,10 @@ def main_const_eval_38(arg):
     return [ttnn_to_device_74]
 
 
-def consteval__main(ce_cache, weights):
+def consteval__main(ce_cache, weights, device):
     if not ce_cache:
         main_const_eval_0_0 = main_const_eval_0(
-            [weights["L__self___model_embed_tokens.weight"]]
+            [weights["L__self___model_embed_tokens.weight"]], device
         )
         ce_cache["main_const_eval_0"] = main_const_eval_0_0[0]
         main_const_eval_1_0 = main_const_eval_1(
@@ -1781,11 +1741,11 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_20_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_20_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_20_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_1"] = main_const_eval_1_0[0]
         main_const_eval_2_0 = main_const_eval_2(
-            [weights["L__self___model_rotary_emb_inv_freq"]]
+            [weights["L__self___model_rotary_emb_inv_freq"]], device
         )
         ce_cache["main_const_eval_2"] = [main_const_eval_2_0[0], main_const_eval_2_0[1]]
         main_const_eval_3_0 = main_const_eval_3(
@@ -1793,7 +1753,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_24_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_24_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_24_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_3"] = main_const_eval_3_0[0]
         main_const_eval_4_0 = main_const_eval_4(
@@ -1801,7 +1761,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_16_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_16_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_16_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_4"] = main_const_eval_4_0[0]
         main_const_eval_5_0 = main_const_eval_5(
@@ -1809,7 +1769,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_3_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_3_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_3_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_5"] = main_const_eval_5_0[0]
         main_const_eval_6_0 = main_const_eval_6(
@@ -1817,7 +1777,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_1_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_1_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_1_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_6"] = main_const_eval_6_0[0]
         main_const_eval_7_0 = main_const_eval_7(
@@ -1825,7 +1785,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_26_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_26_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_26_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_7"] = main_const_eval_7_0[0]
         main_const_eval_8_0 = main_const_eval_8(
@@ -1833,7 +1793,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_18_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_18_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_18_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_8"] = main_const_eval_8_0[0]
         main_const_eval_9_0 = main_const_eval_9(
@@ -1841,7 +1801,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_28_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_28_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_28_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_9"] = main_const_eval_9_0[0]
         main_const_eval_10_0 = main_const_eval_10(
@@ -1849,7 +1809,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_35_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_35_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_35_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_10"] = main_const_eval_10_0[0]
         main_const_eval_11_0 = main_const_eval_11(
@@ -1857,7 +1817,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_8_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_8_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_8_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_11"] = main_const_eval_11_0[0]
         main_const_eval_12_0 = main_const_eval_12(
@@ -1865,7 +1825,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_9_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_9_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_9_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_12"] = main_const_eval_12_0[0]
         main_const_eval_13_0 = main_const_eval_13(
@@ -1873,7 +1833,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_5_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_5_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_5_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_13"] = main_const_eval_13_0[0]
         main_const_eval_14_0 = main_const_eval_14(
@@ -1881,7 +1841,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_15_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_15_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_15_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_14"] = main_const_eval_14_0[0]
         main_const_eval_15_0 = main_const_eval_15(
@@ -1889,7 +1849,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_2_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_2_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_2_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_15"] = main_const_eval_15_0[0]
         main_const_eval_16_0 = main_const_eval_16(
@@ -1897,7 +1857,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_12_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_12_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_12_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_16"] = main_const_eval_16_0[0]
         main_const_eval_17_0 = main_const_eval_17(
@@ -1905,7 +1865,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_22_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_22_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_22_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_17"] = main_const_eval_17_0[0]
         main_const_eval_18_0 = main_const_eval_18(
@@ -1913,7 +1873,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_25_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_25_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_25_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_18"] = main_const_eval_18_0[0]
         main_const_eval_19_0 = main_const_eval_19(
@@ -1921,7 +1881,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_0_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_0_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_0_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_19"] = main_const_eval_19_0[0]
         main_const_eval_20_0 = main_const_eval_20(
@@ -1929,10 +1889,10 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_23_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_23_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_23_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_20"] = main_const_eval_20_0[0]
-        main_const_eval_21_0 = main_const_eval_21()
+        main_const_eval_21_0 = main_const_eval_21(device)
         ce_cache["main_const_eval_21"] = [
             main_const_eval_21_0[0],
             main_const_eval_21_0[1],
@@ -1976,7 +1936,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_19_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_19_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_19_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_22"] = main_const_eval_22_0[0]
         main_const_eval_23_0 = main_const_eval_23(
@@ -1984,7 +1944,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_29_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_29_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_29_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_23"] = main_const_eval_23_0[0]
         main_const_eval_24_0 = main_const_eval_24(
@@ -1992,7 +1952,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_32_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_32_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_32_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_24"] = main_const_eval_24_0[0]
         main_const_eval_25_0 = main_const_eval_25(
@@ -2000,7 +1960,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_31_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_31_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_31_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_25"] = main_const_eval_25_0[0]
         main_const_eval_26_0 = main_const_eval_26(
@@ -2008,7 +1968,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_6_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_6_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_6_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_26"] = main_const_eval_26_0[0]
         main_const_eval_27_0 = main_const_eval_27(
@@ -2016,7 +1976,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_21_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_21_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_21_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_27"] = main_const_eval_27_0[0]
         main_const_eval_28_0 = main_const_eval_28(
@@ -2024,7 +1984,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_10_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_10_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_10_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_28"] = main_const_eval_28_0[0]
         main_const_eval_29_0 = main_const_eval_29(
@@ -2032,7 +1992,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_11_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_11_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_11_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_29"] = main_const_eval_29_0[0]
         main_const_eval_30_0 = main_const_eval_30(
@@ -2040,7 +2000,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_13_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_13_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_13_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_30"] = main_const_eval_30_0[0]
         main_const_eval_31_0 = main_const_eval_31(
@@ -2048,7 +2008,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_14_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_14_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_14_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_31"] = main_const_eval_31_0[0]
         main_const_eval_32_0 = main_const_eval_32(
@@ -2056,7 +2016,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_33_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_33_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_33_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_32"] = main_const_eval_32_0[0]
         main_const_eval_33_0 = main_const_eval_33(
@@ -2064,7 +2024,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_34_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_34_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_34_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_33"] = main_const_eval_33_0[0]
         main_const_eval_34_0 = main_const_eval_34(
@@ -2072,7 +2032,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_4_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_4_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_4_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_34"] = main_const_eval_34_0[0]
         main_const_eval_35_0 = main_const_eval_35(
@@ -2080,7 +2040,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_17_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_17_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_17_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_35"] = main_const_eval_35_0[0]
         main_const_eval_36_0 = main_const_eval_36(
@@ -2088,7 +2048,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_30_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_30_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_30_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_36"] = main_const_eval_36_0[0]
         main_const_eval_37_0 = main_const_eval_37(
@@ -2096,7 +2056,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_7_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_7_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_7_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_37"] = main_const_eval_37_0[0]
         main_const_eval_38_0 = main_const_eval_38(
@@ -2104,7 +2064,7 @@ def consteval__main(ce_cache, weights):
                 weights["L__self___model_layers_27_self_attn_v_proj.weight"],
                 weights["L__self___model_layers_27_self_attn_k_proj.weight"],
                 weights["L__self___model_layers_27_self_attn_q_proj.weight"],
-            ]
+            ], device
         )
         ce_cache["main_const_eval_38"] = main_const_eval_38_0[0]
     return ce_cache
diff --git a/Qwen/Qwen3-4B/model/graph_0/main.py b/Qwen/Qwen3-4B/model/graph_0/main.py
index 0df57fa..8cdc1fc 100644
--- a/Qwen/Qwen3-4B/model/graph_0/main.py
+++ b/Qwen/Qwen3-4B/model/graph_0/main.py
@@ -8,9 +8,9 @@ from utils import calculate_pcc
 ce_cache__main = {}
 
 
-def _main(activations, weights):
+def _main(activations, weights, device):
     global ce_cache__main
-    ce_cache__main = consteval__main(ce_cache__main, weights)
+    ce_cache__main = consteval__main(ce_cache__main, weights, device)
     args_0 = activations[0]
     var_0 = ce_cache__main["main_const_eval_2"]
     var_1 = var_0[0]
@@ -25126,13 +25126,12 @@ def _main(activations, weights):
     ]
 
 
-def load_activations_for__main():
-    utils_DeviceGetter_get_device_0 = utils.DeviceGetter.get_device((1, 1))
+def load_activations_for__main(device):
     utils_load_tensor_0 = utils.load_tensor(
         "./tensors/arg2.tensorbin",
         ttnn.Layout.ROW_MAJOR,
         ttnn.DataType.INT32,
-        utils_DeviceGetter_get_device_0,
+        device,
         ttnn.MemoryConfig(
             ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
         ),
@@ -25141,38 +25140,54 @@ def load_activations_for__main():
 
 
 def main():
-    load_activations_for__main_0 = load_activations_for__main()
-    load_weights_for__main_0 = load_weights_for__main_from_state_dict()
-    _main_0 = _main(load_activations_for__main_0, load_weights_for__main_0)
+    device = ttnn.open_mesh_device(
+        mesh_shape=ttnn.MeshShape((1, 1)),
+        l1_small_size=1 << 15,
+    )
+    print(f"Device: {device}")
+    try:
+        load_activations_for__main_0 = load_activations_for__main(device)
+        load_weights_for__main_0 = load_weights_for__main_from_state_dict(device)
+        _main_0 = _main(load_activations_for__main_0, load_weights_for__main_0, device)
+    finally:
+        ttnn.close_mesh_device(device)
+        ttnn.set_fabric_config(ttnn.FabricConfig.DISABLED)
     return 0
 
 
 def test_main():
     exact_pcc = 0.9765207171440125
 
-    device = utils.DeviceGetter.get_device((1, 1))
-
-    input = model_pt.load_input()
-    input = ttnn.from_torch(input)
-    input = ttnn.to_layout(input, ttnn.Layout.ROW_MAJOR)
-    input = ttnn.to_dtype(input, ttnn.DataType.INT32)
-    input = ttnn.to_device(
-        input,
-        device,
-        ttnn.MemoryConfig(
-            ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
-        ),
-    )
+    device = ttnn.open_mesh_device(
+        mesh_shape=ttnn.MeshShape((1, 1)),
+        l1_small_size=1 << 15,
+    )
+    print(f"Device: {device}")
+    try:
+        input = model_pt.load_input()
+        input = ttnn.from_torch(input)
+        input = ttnn.to_layout(input, ttnn.Layout.ROW_MAJOR)
+        input = ttnn.to_dtype(input, ttnn.DataType.INT32)
+        input = ttnn.to_device(
+            input,
+            device,
+            ttnn.MemoryConfig(
+                ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None
+            ),
+        )
 
-    weights = load_weights_for__main_from_state_dict()
-    outputs = _main([input], weights)
+        weights = load_weights_for__main_from_state_dict(device)
+        outputs = _main([input], weights, device)
 
-    ttnn_output = ttnn.to_torch(ttnn.from_device(outputs[-1]))
-    golden_output = model_pt.run_pytorch_model()
+        ttnn_output = ttnn.to_torch(ttnn.from_device(outputs[-1]))
+        golden_output = model_pt.run_pytorch_model()
 
-    pcc = calculate_pcc(ttnn_output, golden_output)
-    print(f"\nPCC: {pcc:.6f}")
-    assert pcc >= exact_pcc, f"PCC {pcc} is not greater than threshold {exact_pcc}"
+        pcc = calculate_pcc(ttnn_output, golden_output)
+        print(f"\nPCC: {pcc:.6f}")
+        assert pcc >= exact_pcc, f"PCC {pcc} is not greater than threshold {exact_pcc}"
+    finally:
+        ttnn.close_mesh_device(device)
+        ttnn.set_fabric_config(ttnn.FabricConfig.DISABLED)
 
 
 if __name__ == "__main__":
diff --git a/Qwen/Qwen3-4B/model/graph_0/params.py b/Qwen/Qwen3-4B/model/graph_0/params.py
index 2d37dd1..bf1f2b2 100644
--- a/Qwen/Qwen3-4B/model/graph_0/params.py
+++ b/Qwen/Qwen3-4B/model/graph_0/params.py
@@ -4,7 +4,6 @@
 
 import ttnn
 import model_pt
-import utils
 
 
 ROW_MAJOR_BFLOAT16_WEIGHTS = {
@@ -437,9 +436,7 @@ def _state_dict_key(weight_name):
     return name
 
 
-def load_weights_for__main_from_state_dict():
-    device = utils.DeviceGetter.get_device((1, 1))
-
+def load_weights_for__main_from_state_dict(device):
     model = model_pt.load_pytorch_model()
     state_dict = dict(model.state_dict())
     for name, buf in model.named_buffers():
diff --git a/Qwen/Qwen3-4B/model/graph_0/utils.py b/Qwen/Qwen3-4B/model/graph_0/utils.py
index 1fff9c6..7da65fc 100644
--- a/Qwen/Qwen3-4B/model/graph_0/utils.py
+++ b/Qwen/Qwen3-4B/model/graph_0/utils.py
@@ -2,7 +2,6 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 import ttnn
-import math
 import torch
 
 
@@ -26,49 +25,6 @@ def calculate_pcc(x, y):
     return float("nan") if denom == 0 else ((vx @ vy) / denom).item()
 
 
-class DeviceGetter:
-    _instance = None
-    _mesh_shape = None
-    l1_small_size = 1 << 15
-
-    def __init__(self):
-        raise RuntimeError("This is Singleton, invoke get_device() instead.")
-
-    def __del__(self):
-        if self._instance is not None:
-            ttnn.close_mesh_device(self._instance)
-            ttnn.set_fabric_config(ttnn.FabricConfig.DISABLED)
-
-    @classmethod
-    def get_device(cls, mesh_shape):
-        if cls._instance == None:
-            if (
-                not isinstance(mesh_shape, (list, tuple))
-                or len(mesh_shape) == 0
-                or not all(isinstance(x, int) and x > 0 for x in mesh_shape)
-            ):
-                raise ValueError(
-                    f"mesh_shape must be a non-empty list or tuple of positive integers, got {mesh_shape}"
-                )
-            cls._mesh_shape = mesh_shape
-
-            if math.prod(mesh_shape) >= 2:
-                ttnn.set_fabric_config(ttnn.FabricConfig.FABRIC_1D)
-            cls._instance = ttnn.open_mesh_device(
-                mesh_shape=ttnn.MeshShape(mesh_shape),
-                l1_small_size=cls.l1_small_size,
-            )
-            print(f"Device: {cls._instance}")
-
-        # Compare requested mesh_shape with _mesh_shape used to initialize the device
-        if tuple(cls._mesh_shape) != tuple(mesh_shape):
-            raise ValueError(
-                f"Device already initialized with mesh_shape={cls._mesh_shape}, but got mesh_shape={mesh_shape}"
-            )
-
-        return cls._instance
-
-
 def get_scalar_from_tensor(tensor: ttnn.Tensor) -> int:
     assert tensor.logical_volume() == 1, "expected scalar tensor"
     assert tensor.dtype == ttnn.DataType.UINT32, "expected uint32 tensor"
