diff --git a/meta-llama/Llama-3.1-8B-Instruct/model/graph_0/main.py b/meta-llama/Llama-3.1-8B-Instruct/model/graph_0/main.py
index f0f3e5b..ec11679 100644
--- a/meta-llama/Llama-3.1-8B-Instruct/model/graph_0/main.py
+++ b/meta-llama/Llama-3.1-8B-Instruct/model/graph_0/main.py
@@ -2,6 +2,7 @@ import ttnn
 import utils
 import ttir_cpu
 import torch
+import model_pt
 from utils import calculate_pcc
 
 
@@ -33569,7 +33570,20 @@ def main():
 
 
 def test_main():
-    return 0
+    exact_pcc = 0.9921875
+
+    activations = load_activations_for__main()
+    weights = load_weights_for__main()
+    outputs = _main(activations, weights)
+
+    ttnn_output = ttnn.from_device(outputs[-1])
+    ttnn_output = ttnn.to_torch(ttnn_output).to(torch.float32).reshape(32, 128256)
+
+    golden_output = model_pt.run_pytorch_model().to(torch.float32)
+
+    pcc = calculate_pcc(ttnn_output, golden_output)
+    print(f"\nPCC: {pcc:.6f}")
+    assert pcc == exact_pcc, f"PCC {pcc} does not match expected {exact_pcc}"
 
 
 if __name__ == "__main__":
